### PR TITLE
add value color

### PR DESCRIPTION
**Pull request by dudjima**
add input color in an event.

### CODE REVIEW

File: lib/GaletteEvents/Event.php [hunk 1]
```diff
@@ -60,6 +60,7 @@ class Event
     private bool $open = true;
     private ?int $group;
     private string $comment = '';
+    private strnig $color;
```
Comment (trasher) [inline lib/GaletteEvents/Event.php]:
```suggestion
    private string $color;
```

**Review by trasher — COMMENTED** [review]
What it the goal? That color is just stored in database, and never used as far as I've seen.

**Comment by dudjima** [comment]
fullcalendar takes color value into calendar display

**Comment by trasher** [comment]
> fullcalendar takes color value into calendar display

Hum... Okay, but I think there is something missing to achieve that ;)

**Comment by trasher** [comment]
> > fullcalendar takes color value into calendar display
> 
> Hum... Okay, but I think there is something missing to achieve that ;)

Ping?

**Comment by dudjima** [comment]
sorry, i didn’t see you're message. 

no other update, it work just with a small change add input color, update database and udpate object event with color value.

**Comment by trasher** [comment]
Okay, indeed, this is automatically passed to calendar.

I've just tested, I had to made changes for storage to work; and I've added update scripts; all seems to go well.

From an UI point of view, there is an issue when background color is set to a light color, since text is white:
image

I have no idea how to get that fixed for now

**Comment by trasher** [comment]
I had something that seems correct with the following changes:
```diff
diff --git a/lib/GaletteEvents/Repository/Events.php b/lib/GaletteEvents/Repository/Events.php
index 452c38a..8f1bb3f 100644
--- a/lib/GaletteEvents/Repository/Events.php
+++ b/lib/GaletteEvents/Repository/Events.php
@@ -192,6 +192,7 @@ class Events
                     $end_date = new \DateTime($event->getEndDate(false));
                     if ($fullcalendar === true) {
                         $end_date = $end_date->modify('+1 day');
+                        $row['textColor'] = $this->getForegoundColor($row['color'] ?? '#ffffff');
                     }
                     $row['end'] = $end_date->format('Y-m-d');
 
@@ -385,4 +386,21 @@ class Events
     {
         return $this->count;
     }
+
+    /**
+     * Get foreground contrasted color for a background color
+     *
+     * @param string $bgcolor Background color in hex format
+     *
+     * @return string
+     */
+    protected function getForegoundColor(string $bgcolor): string
+    {
+        $bgcolor = trim($bgcolor, '#');
+        $r = hexdec(substr($bgcolor,0,2));
+        $g = hexdec(substr($bgcolor,2,2));
+        $b = hexdec(substr($bgcolor,4,2));
+        $yiq = (($r*299)+($g*587)+($b*114))/1000;
+        return ($yiq >= 128) ? 'black' : 'white';
+    }
 }
```

**Comment by trasher** [comment]
image

**Review by trasher — APPROVED** [review]

**Pull request merged**